nvcodec/samples/simple/SampleEncodeDecode.java

```diff
@@ -28,25 +28,30 @@ public static void main(String[] args) {
 
         checkCudaApiCall("cuInit", cuInit(0));
         checkCudaApiCall("cuCtxCreate", cuCtxCreate(cuContext, 0, targetGpu));
-        // Check encoder max supported version
-        {
-            IntPointer version = new IntPointer(1);
-
-            checkEncodeApiCall("NvEncodeAPIGetMaxSupportedVersion", NvEncodeAPIGetMaxSupportedVersion(version));
-
-            System.out.printf("Encoder Max Supported Version\t : %d \r\n", version.get());
-        }
-
-        // Query decoder capability 'MPEG-1' codec
-        {
-            CUVIDDECODECAPS decodeCaps = new CUVIDDECODECAPS();
-            decodeCaps.eCodecType(cudaVideoCodec_HEVC);
-            decodeCaps.eChromaFormat(cudaVideoChromaFormat_420);
-            decodeCaps.nBitDepthMinus8(2); // 10 bit
-
-            checkCudaApiCall("cuvidGetDecoderCaps", cuvidGetDecoderCaps(decodeCaps));
-
-            System.out.printf("Decoder Capability MPEG-1 Codec\t : %s \r\n", (decodeCaps.bIsSupported() != 0));
+        try {
+            // Check encoder max supported version
+            {
+                try (IntPointer version = new IntPointer(1)) {
+                    checkEncodeApiCall("NvEncodeAPIGetMaxSupportedVersion", NvEncodeAPIGetMaxSupportedVersion(version));
+
+                    System.out.printf("Encoder Max Supported Version\t : %d \r\n", version.get());
+                }
+            }
+
+            // Query decoder capability 'H264' codec
+            {
+                try (CUVIDDECODECAPS decodeCaps = new CUVIDDECODECAPS()) {
+                    decodeCaps.eCodecType(cudaVideoCodec_H264);
+                    decodeCaps.eChromaFormat(cudaVideoChromaFormat_420);
+                    decodeCaps.nBitDepthMinus8(0);
+
+                    checkCudaApiCall("cuvidGetDecoderCaps", cuvidGetDecoderCaps(decodeCaps));
+
+                    System.out.printf("Decoder Capability H264 Codec\t : %s \r\n", (decodeCaps.bIsSupported() != 0));
+                }
+            }
+        } finally {
```

Comment:
I don't think we need a `finally` here since there's nothing that throws from the `try { }` block, or is there?

Reply:
I added excepthon throws in `checkEncodeApiCall` and `checkCudaApiCall`!